src/representative_subset_calculator/streaming/candidate_consumer.h

```diff
@@ -159,4 +158,156 @@ class SeiveCandidateConsumer : public CandidateConsumer {
 
         return highestMarginal;
     }
+};
+
+
+class ThreeSeiveCandidateConsumer : public CandidateConsumer {
+    private:
+    const unsigned int numberOfSenders;
+    const unsigned int k;
+    const double epsilon;
+    unsigned int T; // hyperparameter -- tranfer contents to next bucket after T unsuccesful inserts
+    unsigned int t; // counts till T
+    unsigned int currentBucketIndex;
+
+    std::vector<std::unique_ptr<CandidateSeed>> seedStorage;
+    std::unordered_set<unsigned int> seenFirstElement;
+    std::unique_ptr<std::vector<ThresholdBucket>> buckets = std::make_unique<std::vector<ThresholdBucket>>();
```

Comment:
Also, as seen in the other comments, you may want to just maintain a single unique pointer to a single bucket instead of a vector of pointers. Maybe instead of `std::vector<std::unique_ptr<ThresholdBucket>>` you do `std::unique_ptr<ThresholdBucket>`